hibernate-reactive-core/src/main/java/org/hibernate/reactive/persister/entity/impl/ReactiveAbstractEntityPersister.java

```diff
@@ -501,16 +500,15 @@ else if ( isAllOrDirtyOptimisticLocking() && loadedState != null ) {
 		}
 
 		Object[] state = loadedState;
-		return loop(
-				IntStream.iterate(span-1, (i) -> i-1 ).limit( span ),
-				table -> deleteReactive(
-						id,
-						version,
-						table,
-						deleteStrings[table],
-						session,
-						state
-				)
+		return loop( 0, span,
+					 table -> deleteReactive(
+							 id,
+							 version,
+							 span - table - 1,
+							 deleteStrings[span - table - 1],
+							 session,
+							 state
+					 )
```

Comment:
But doesn't this reverse the order of the iteration?

Reply:
That was what the original code was doing

Reply:
Oh, apologies, I missed that you had changed it to use `span - table - 1`. That's better, actually.